.github/workflows/CI_Pipeline.yml

```diff
@@ -26,19 +26,22 @@ jobs:
     - run: docker build --target test --tag my-test-image .
     - run: docker run --env-file .env.test  my-test-image tests
     - run: docker run -e SECRET_KEY=secret -e MONGO_DB_NAME=e2e_db -e MONGO_CONNECT=${{secrets.MONGO_CONNECT}} -e MONGO_COLLECTION_NAME=${{secrets.MONGO_COLLECTION_NAME}} -e MONGO_DEFAULT_STATUS=${{secrets.MONGO_DEFAULT_STATUS}} my-test-image tests_e2e
-##Commenting out as still causing issues
-#  deploy:
-#    name: Deploy build package to Heroku
-#    needs: build
-#    if: ${{ github.event_name ==  'push' }}
-#    runs-on: ubuntu-latest
-#    steps:
-#    - uses: actions/checkout@v2 # Adds a step to checkout the repository code
-    #This action was used for simplicity and conciseness however this action has the inability to specify build options
-#    - uses: akhileshns/heroku-deploy@v3.12.12 # This is the action 
-#      with:
-#        heroku_api_key: ${{secrets.HEROKU_API_KEY}}
-#        heroku_app_name: "joe-devops-project" #Must be unique in Heroku
-#        heroku_email: "josephbwo@hotmail.com"
-#        usedocker: true
-        
\ No newline at end of file
+    - run: docker logout
+  publish:
+    name: Push Image to dockerhub
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: docker/login-action@v2
+      with:
+        username: ${{secrets.DOCKERHUB_USERNAME}}
+        password: ${{secrets.DOCKERHUB_TOKEN}}
+    - run: docker build --target production --tag joeosbo/devopsproj:todo_app .    
+    - run: docker logout
+    - uses: docker/login-action@v2
+      with:
+        username: ${{secrets.DOCKERHUB_USERNAME}}
+        password: ${{secrets.DOCKERHUB_TOKEN}}
+    - run: docker push joeosbo/devopsproj:todo_app 
+    - run: curl -dH -X POST ${{secrets.AZURE_WEBHOOK}}
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400):